File: examples/java/micro/UserService.java
package micro;

import java.util.HashMap;
import java.util.Map;

public class UserService {
    private AuthService authService;

    public UserService() throws Exception {
        this.authService = new AuthService();
    }

    public Map<String, String> getUser(String token) throws Exception {
        String userId = authService.verifyToken(token);
        Map<String, String> user = new HashMap<>();
        user.put("id", userId);
        user.put("token", token);
        return user;
    }

    public String updateUser(String token, String action) throws Exception {
        String userId = authService.verifyToken(token);
        if (userId != null) {
            authService.runAdminCommand(userId, action);
        }
        return userId;
    }
}

File: test-fixtures/languages/go/micro/user_service.go
package micro

import "database/sql"

type UserInfo struct {
	ID    string
	Token string
}

func GetUser(db *sql.DB, token string) (*UserInfo, error) {
	userID, err := VerifyToken(db, token)
	if err != nil {
		return nil, err
	}
	return &UserInfo{ID: userID, Token: token}, nil
}

func UpdateUser(db *sql.DB, token string, action string) (string, error) {
	userID, err := VerifyToken(db, token)
	if err != nil {
		return "", err
	}
	if userID != "" {
		err = RunAdminCommand(userID, action)
	}
	return userID, err
}
